### PR TITLE
Gherkin script for Pipelines - Improve Pipeline Creation Process- ODC…

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-3925
Story: https://issues.redhat.com/browse/ODC-4825

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Feature: Pipeline features in topology
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:1]: Can you please update the feature with some more information?

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	As a user, I want to check pipeline features in topology
+
+Background:
+    Given user is at developer perspecitve
+    And user has installed pipeline operator
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:6]:
```suggestion
    And user has installed OpenShift Pipelines Operator
```

**Review by gajanan-more — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:1]: It's regarding pipelines in topology

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:6]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 3]
```diff
@@ -0,0 +1,29 @@
+Feature: Pipeline features in topology
+	As a user, I want to check pipeline features in topology
+
+Background:
+    Given user is at developer perspecitve
+    And user has installed Openshift Pipeline Operator
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:6]:
```suggestion
    And user has installed Openshift Pipelines Operator
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 4]
```diff
@@ -0,0 +1,29 @@
+Feature: Pipeline features in topology
+	As a user, I want to check pipeline features in topology
+
+Background:
+    Given user is at developer perspecitve
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:5]:
```suggestion
    Given user is at developer perspective
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    | deployment config | dancer-ex-git-1 |
+
+
+@regression
+Scenario Outline: Sidebar for workload
+   Given user created workload "<workload_name>" with resource type "<resource_type>" with pipeline 
+   When user clicks on workload "<workload_name>"
+   Then user can see sidebar opens with Resources tab selected by default
+   And user can see Pipeline Runs section
+   And user is able to see Start button to start a pipeline run
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:21]: Please change the scenario based on the below pointers.

1. When a pipeline options is checked in during the Import flow, then in the topology sidebar user should be able to see then pipelinerun section but the buildconfig section should **NOT** be available.
2. During the +Add flow creation, if the pipeline is not able to auto-start for some reason, only then we will be showing the start button  in the sidebar.

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+
+@regression @manual
+Scenario Outline: Pipeline icon on application node
+   Given user created workload "<workload_name>" with resource type "<resource_type>" with pipeline 
+   When user hovers on the bottom-left decorator on the workload "<workload_name>"
+   Then user can see Pipeline not started text
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:28]: On clicking the `not started icon` should open the pipeline start modal.

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 7]
```diff
@@ -0,0 +1,31 @@
+Feature: Pipeline features in topology
+	As a user, I want to check pipeline features in topology
+
+Background:
+    Given user is at developer perspective
+    And user has installed Openshift Pipelines   Operator
```
Comment (gajanan-more) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:6]:
```suggestion
    And user has installed OpenShift Pipelines Operator
```

**Review by gajanan-more — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    And user has selected namespace "aut-topology-pipeline"
+
+Examples:
+    | resource_type     | workload_name   |
+    | deployment        | nodejs-ex-git-1 |
+    | deployment config | dancer-ex-git-1 |
+
+
+@regression
+Scenario Outline: Sidebar for workload
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:16]:
```suggestion
Scenario: Workload Sidebar
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+   Given user created workload "<workload_name>" with resource type "<resource_type>" with pipeline
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:17]:
```suggestion
   Given user created workload "pipeline-test" with resource type "deployment" with pipeline 
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+Background:
+    Given user is at developer perspective
+    And user has installed Openshift Pipelines Operator
+    And user has selected namespace "aut-topology-pipeline"
+
+Examples:
+    | resource_type     | workload_name   |
+    | deployment        | nodejs-ex-git-1 |
+    | deployment config | dancer-ex-git-1 |
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:12]: Remove here

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+@regression
+Scenario Outline: Sidebar for workload
+   Given user created workload "<workload_name>" with resource type "<resource_type>" with pipeline 
+   And pipeline has not autostarted
+   When user clicks on workload "<workload_name>"
+   Then user can see sidebar opens with Resources tab selected by default
+   And user can see Pipeline Runs section
+   And user can not see Buildconfig Runs section   
+   And user is able to see Start button to start a pipeline run
+
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:24]:
```suggestion
Examples:
    | resource_type     | workload_name   |
    | deployment        | nodejs-ex-git-1 |
    | deployment config | dancer-ex-git-1 |
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+@regression @manual
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:26]:
```suggestion
@regression,  @manual
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+Scenario Outline: Pipeline icon on application node
+   Given user created workload "<workload_name>" with resource type "<resource_type>" with pipeline 
+   When user hovers on the bottom-left decorator on the workload "<workload_name>"
+   Then user can see Pipeline not started text which on clicking leads to pipeline start model
+
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:31]:
```suggestion
Examples:
    | resource_type     | workload_name   |
    | deployment        | nodejs-ex-git-1 |
    | deployment config | dancer-ex-git-1 |
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+Examples:
+    | resource_type     | workload_name   |
+    | deployment        | nodejs-ex-git-1 |
+    | deployment config | dancer-ex-git-1 |
+
+
+@regression
+Scenario Outline: Sidebar for workload
+   Given user created workload "<workload_name>" with resource type "<resource_type>" with pipeline 
+   And pipeline has not autostarted
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:18]:
```suggestion
   And user is redirected to topology page
   And if pipeline failed to started automatically
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 8 above)
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:16]: I think scenario description should be different [Think as per the https://issues.redhat.com/browse/ODC-3925 criteria]

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+   When user clicks on workload "<workload_name>"
+   Then user can see sidebar opens with Resources tab selected by default
+   And user can see Pipeline Runs section
+   And user can not see Buildconfig Runs section
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:22]:
```suggestion
    And user can see an pipeline not started alert 
    And user can not see Buildconfig Runs section   
```

Comment (karthikjeeyar) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:22]: Refer https://github.com/openshift/console/pull/7241 for more details

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+   Then user can see sidebar opens with Resources tab selected by default
+   And user can see Pipeline Runs section
+   And user can not see Buildconfig Runs section   
+   And user is able to see Start button to start a pipeline run
+
+
+@regression @manual
+Scenario Outline: Pipeline icon on application node
+   Given user created workload "<workload_name>" with resource type "<resource_type>" with pipeline 
+   When user hovers on the bottom-left decorator on the workload "<workload_name>"
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:29]:
```suggestion
   When user hovers on the bottom-left decorator on the workload "<workload_name>"
   And user clicks on the pipeline with not started text
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+   And pipeline has not autostarted
+   When user clicks on workload "<workload_name>"
+   Then user can see sidebar opens with Resources tab selected by default
+   And user can see Pipeline Runs section
+   And user can not see Buildconfig Runs section   
+   And user is able to see Start button to start a pipeline run
+
+
+@regression @manual
+Scenario Outline: Pipeline icon on application node
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:27]:
```suggestion
Scenario Outline: Pipeline start modal on clicking pipeline icon of application node in topology page
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+   Given user created workload "<workload_name>" with resource type "<resource_type>" with pipeline 
+   When user hovers on the bottom-left decorator on the workload "<workload_name>"
+   Then user can see Pipeline not started text which on clicking leads to pipeline start model
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:30]:
```suggestion
   Then user can see Pipeline start model
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 1 above)
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:1]: It should be more descriptive or please add this to the existing feature files

**Review by makambalaji — COMMENTED** [review]
@sanketpathak  - Please look at my comments once

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 14 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:18]: if can not be used in the script

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 11 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:19]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 12 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:26]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 13 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:32]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 8 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 15 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:16]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 16 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:41]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 17 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 18 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 (hunk 10 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:21]: Updated

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 19]
```diff
@@ -0,0 +1,167 @@
+Feature: Improve the integration of Pipelines & Builds.
+	As a user, I want to see integration of pipelines and builds
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:2]: This feature intends to either use pipeline or builds, not both at a time.

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+    | deployment        | s-2-i-dotnetcore-ex-git-1 | .Net core     |
+    | deployment config | dancer-ex-git-1           | Perl          |
+
+
+@regression, @manual
+Scenario Outline: Pipeline start modal on clicking pipeline icon of application node in topology page
+   Given user created workload "<workload_name>" with resource type "<resource_type>" with pipeline 
+   When user hovers on the bottom-left decorator on the workload "<workload_name>"
+   And user clicks on the pipeline with not started text
+   Then user can see Pipeline start model
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:43]:
```suggestion
   Then user can see Pipeline start modal
```

File: frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925 [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+    | deployment        | nodejs-ex-git-1 |
+    | deployment config | dancer-ex-git-1 |
+
+
+@regression
+Scenario Outline: Add flow should finish even if the Pipeline Run cannot be created for the added Pipeline
+   Given user created workload "<workload_name>" with resource type "<resource_type>" and builder image "<builder_image>" with pipeline 
+   And pipeline has not autostarted
+   When user clicks on workload "<workload_name>"
+   Then user can see alert "Pipeline could not be started automatically" in the  Pipeline Runs section
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/topology/topology_pipeline_feature-ODC-3925:29]:
```suggestion
   Then user can see an alert "Pipeline could not be started automatically" in the Pipeline Runs section of the sidebar
```

**Comment by rottencandy** [comment]
:+1:

**Comment by karthikjeeyar** [comment]
Thanks @sanketpathak  
/lgtm

**Pull request merged**